include/seqan3/search/algorithm/search.hpp

```diff
@@ -46,15 +74,29 @@ namespace seqan3
  * Strong exception guarantee if iterating the query does not change its state and if invoking a possible delegate
  * specified in `cfg` also has a strong exception guarantee; basic exception guarantee otherwise.
  */
-template <FmIndex index_t, typename queries_t, typename configuration_t>
-//!\cond
-    requires
-        (std::ranges::RandomAccessRange<queries_t> ||
-            (std::ranges::ForwardRange<queries_t> && std::ranges::RandomAccessRange<value_type_t<queries_t>>)) &&
-        detail::is_type_specialisation_of_v<remove_cvref_t<configuration_t>, configuration>
-//!\endcond
-inline auto search(queries_t && queries, index_t const & index, configuration_t const & cfg)
+template <FmIndex index_t, typename queries_t, typename configuration_t = decltype(search_cfg::default_configuration)>
+inline auto search(queries_t && queries,
+                   index_t const & index,
+                   configuration_t const & cfg = search_cfg::default_configuration)
 {
+    if constexpr(dimension_v<queries_t> == 1u)
+    {
+        static_assert(std::ranges::RandomAccessRange<queries_t>, "The query sequence must model RandomAccessRange.");
+        static_assert(std::ranges::SizedRange<queries_t>, "The query sequence must model SizedRange.");
+    }
+    else
+    {
+        static_assert(std::ranges::ForwardRange<queries_t>, "The query range must model ForwardRange.");
+        static_assert(std::ranges::SizedRange<queries_t>, "The query range must model SizedRange.");
+        static_assert(std::ranges::RandomAccessRange<value_type_t<queries_t>>,
+                      "Elements of the query range must model RandomAccessRange.");
```

Comment:
call it maybe collection instead of range as the query itself is also a range and it could be misleading 💅

Reply:
All four asserts or only the last two?

Reply:
Like whenver you mean the collection and not the single query